subsys/bluetooth/mesh/adv_legacy.c

```diff
@@ -178,9 +177,18 @@ void bt_mesh_adv_update(void)
 	BT_DBG("");
 
 	k_fifo_cancel_wait(&bt_mesh_adv_queue);
+
+#if defined(CONFIG_BT_MESH_RELAY)
+	k_fifo_cancel_wait(&bt_mesh_relay_queue);
+#endif /* CONFIG_BT_MESH_RELAY */
```

Comment:
Should put this inside adv.c as well.

Reply:
Where?

Reply:
I mean, since the polling of the queues are implemented in adv.c, it makes sense to put the cancellation of the polling inside that file as well, as a new `bt_mesh_adv_buf_get_cancel()` function or something

Reply:
Maybe, but adv_multi.c really need cancel fifo, since use k_no_wait.